### ISSUE TITLE
Does not work (REST issue)

### ISSUE THREAD

**Issue by lmaiaux** [ansis-m/CodeWarsPlugin#1]
Hello,

In my case, he plugin version 3.0-SNAPSHOT does not work with Pycharm 2023.1.4 Pro.

When I click on "Train" (or "Train Again") for any kata, a popup window appears saying:

Ups, Something Went Wrong...
Kata setup failed. Reason: something went
wrong with the REST calls to the
codewars.com. Try singing in again!

Any idea to debug or to fix it?

Thanks and Regards,
Lionel

**Comment by ansis-m** [comment]
Have you tried signing-out and sining-in again in the browser window? If that does not work, perhaps closing and reopening the PyCharm could do the trick? It might be that the cookies are outdated.

**Comment by ansis-m** [comment]
Did you install manually or from the marketplace?

**Comment by lmaiaux** [comment]
I installed it from the marketplace.
I already tried to sign-out/sign-in and to relaunch PyCharm without success...

**Comment by ansis-m** [comment]
At this point I have just a free PyCharm. I will try to investigate on a PyCharm Pro to see what is the problem. Thank you for reporting!

**Comment by ansis-m** [comment]
Could you tell me what OS are you on! Perhaps the issue might be associated with that. I wrote the plugin and tested only on Windows so if you are on a Mac, there might be some problem with encoding/decoding.

**Comment by lmaiaux** [comment]
FYI your plugin version 3.0-SNAPSHOT is not compatible with PyCharm 2023.2 Pro.
The error message is :

Plugin 'CodeWars' (version '3.0-SNAPSHOT') is not compatible with the current version of the IDE, because it requires build 231.* or older but the current build is PY-232.8660.197

**Comment by lmaiaux** [comment]
I test it on Windows.
I would like to test in on unix (with both PyCharm Pro and PyCharm Community) but I do not see your plugin any more on the Marketplace.
Did you remove it?

**Comment by ansis-m** [comment]
@lmaiaux Thanks for pointing out the problem. I had to upgrade java from 11 to 17. It is now fixed however, it will take a couple of days until the updated plugin version is in the marketplace. In the meanwhile, you can pull the latest version from the github and istall it manually from the disk as described in the README. Thank you again for pointing out the problem!

**Comment by lmaiaux** [comment]
Thanks but I am not able to build the plugin with IntelliJ 2023.2.1 ultimate.
I get a lot of errors like:

[gradle-intellij-plugin :] Error when resolving dependency: DefaultExternalModuleDependency{group='com.jetbrains.intellij.idea', name='ideaIC', version='2022.1.1', configuration='default'}

Does it mean that you are using IntelliJ 2022.1.1 and it cannot build with a more recent release?

**Comment by ansis-m** [comment]
@lmaiaux The updated version of the plugin (4.0-SNAPSHOT) is now in the marketplace. I installed it in the PyCharm 2023.2 and it works fine for me. I am sorry for inconveniences but I am new to programming and this is my first plugin so there might be some problems. I hope it works for you now.

I did write the plugin in Intellij 2022.3.2. Intellij switched from Java11 to Java17 when moving from 2022 to 2023 so that explains the compatibility issues you had with PyCharm. I think that you definatelly should be able to build it with a newer version of Intellij but you might have to tweak the plugin.xml and build.gradle.kts files and project settings to make sure that the jdk matches.
